scripts/lua2dox.lua

```diff
@@ -201,16 +213,15 @@ local function process_magic(line, generics)
 
   if directive == 'param' then
     for _, type in ipairs(TYPES) do
-      magic = magic:gsub('^param%s+([a-zA-Z_?]+)%s+.*%((' .. type .. ')%)', 'param %1 %2')
-      magic =
-        magic:gsub('^param%s+([a-zA-Z_?]+)%s+.*%((' .. type .. '|nil)%)', 'param %1 %2')
+      magic = magic
+        :gsub('^param%s+([a-zA-Z_?]+)%s+.*%((' .. addPossibleTypeCaptureGroups(type) .. ')%)', 'param %1 %2')
     end
     magic_split = vim.split(magic, ' ', { plain = true })
     type_index = 3
   elseif directive == 'return' then
     for _, type in ipairs(TYPES) do
-      magic = magic:gsub('^return%s+.*%((' .. type .. ')%)', 'return %1')
-      magic = magic:gsub('^return%s+.*%((' .. type .. '|nil)%)', 'return %1')
+      magic = magic
+        :gsub('^return%s+.*%((' .. addPossibleTypeCaptureGroups(type) .. ')%)', 'return %1')
```

Comment:
Doing this with a single Lua pattern won't scale very well. Especially using the `[...]*` trick.

Using multiple patterns is fine as long as each pattern does one specific thing.

I think all this block is doing is removing parenthesis. Could it not be simplified to(?):

```lua
magic = magic:gsub('%(([^)]+)%)', '%1')
```